packages/@icon-magic/distribute/src/distribute-svg.ts

```diff
@@ -27,30 +31,45 @@ export async function distributeSvg(
   const spriteNames: spriteConfig = {};
   for (const icon of icons) {
     LOGGER.debug(`calling distributeSvg on ${icon.iconName}: ${icon.iconPath}`);
-    const assets = getIconFlavorsByType(icon, 'svg');
-    // If icon has a distribute config and it indicates it shouldn't be put
-    // in a sprite
-    if (
+    let assets = getIconFlavorsByType(icon, 'svg');
+
+    // Either the icon config or the variant config can have a distribute config. Possibilities:
+    // 1. icon distribute config allows or does not allow sprite, config is applied to all variants
+    // 2. icon distribute config allows or does not allow sprite _and_ variants also have a distribute
+    // config that may or may not allow sprite
+    // Check if a variant has a distribute config
+    const assetsToAddToSprite = assets.filter(asset => {
+      return shouldAddToSprite(asset);
+    });
+    const iconHasSpriteConfig = !(
       icon.distribute &&
       icon.distribute.svg &&
       !icon.distribute.svg.toSprite
-    ) {
-      // Just copy the files to the output
-      await copyIconAssetSvgs(icon.iconName, assets, outputPath);
-    } else {
+    );
+
+    if (iconHasSpriteConfig || assetsToAddToSprite.length) {
+      if (assetsToAddToSprite.length && !iconHasSpriteConfig) {
+        assets = assetsToAddToSprite;
+      }
+      // TO-DO: Handle case where icon config enables sprite but one asset disables
```

Comment:
Not sure we need this as it's just another way to add a fixed set of assets to the sprite and not all. We can get rid of the TODO